File: C_repo/test_2.c
#include <stdio.h>

typedef struct {
    int x;
    int y;
} Point;

// Pass-by-pointer (modifies original)
void move(Point *p) {
    p->x += 5;
}

// Pass-by-value (modifies a copy)
void move_copy(Point p_val) {
    p_val.x += 100;
    printf("Inside move_copy: %d\n", p_val.x);
}

int main() {
    Point p1 = {10, 20};
    printf("Original: %d\n", p1.x); // 10
    
    move_copy(p1); // Pass a copy
    printf("After move_copy: %d\n", p1.x); // Unchanged: 10
    
    move(&p1); // Pass a pointer
    printf("After move: %d\n", p1.x); // Changed: 15
    return 0;
}

File: Runs/Java_repo_mistral-nemo/Test_2.java
public class Test_2 {

    public static class Point {
      public int x;
      public int y;
    
      public Point() {}
    
      public Point(Point other) {
        this.x = other.x;
        this.y = other.y;
      }
    
      public Point(int x, int y) {
        this.x = x;
        this.y = y;
      }
    }

    public static void move(Point p) {
      p.x += 5;
    }

    public static void move_copy(Point p_val) {
      p_val.x += 100;
      System.out.println("Inside move_copy: " + p_val.x);
    }

    public static void main(String[] args) {
      Point p1 = new Point(10, 20);
      System.out.println("Original: " + p1.x);
      move_copy(p1);
      System.out.println("After move_copy: " + p1.x);
      move(p1);
      System.out.println("After move: " + p1.x);
    }

}